cdap-master/src/main/java/co/cask/cdap/data/tools/AbstractQueueUpgrader.java

```diff
@@ -104,8 +104,8 @@ void upgrade() throws Exception {
               LOG.debug("Adding entry {} -> {} for upgrade",
                         Bytes.toString(entry.getKey()), Bytes.toString(entry.getValue()));
               put.add(QueueEntryRow.COLUMN_FAMILY, entry.getKey(), entry.getValue());
-              mutations.add(put);
             }
+            mutations.add(put);
```

Comment:
yea, AbstractQueueUpgrader was adding the Put to the mutations before the put was fully built.
Changed it to only add it to the mutation, once all changes to the put have been mode (After the for loop).